### ISSUE TITLE
Typegoose compilation error

### ISSUE THREAD

**Issue by T00rk** [typegoose/typegoose#582]
## Versions

- System: linux
- NodeJS: 14.16.1
- Typescript: 3.9.9
- Compiler / Transpiler: tsc
- Typegoose(NPM): 7.6.3
- mongoose: 5.10.18
- mongodb: 0.0.0

## What is the Problem?

I started a new npm package using typegoose. When I compile it I got the following error:

```bash

node_modules/@types/mongoose/index.d.ts:101:37 - error TS2694: Namespace '"/home/t00rk/mif/entities/node_modules/mongodb/mongodb.ts34"' has no exported member 'ObjectID'.

101     type MongooseBuiltIns = mongodb.ObjectID | mongodb.Decimal128 | Date | number | boolean;
                                        ~~~~~~~~

node_modules/@types/mongoose/index.d.ts:174:17 - error TS2694: Namespace '"/home/t00rk/mif/entities/node_modules/mongodb/mongodb.ts34"' has no exported member 'RootQuerySelector'.

174         mongodb.RootQuerySelector<T>;
                    ~~~~~~~~~~~~~~~~~

node_modules/@types/mongoose/index.d.ts:184:50 - error TS2694: Namespace '"/home/t00rk/mif/entities/node_modules/mongodb/mongodb.ts34"' has no exported member 'UpdateQuery'.

184     export type MongooseUpdateQuery<S> = mongodb.UpdateQuery<S> & mongodb.MatchKeysAndValues<S>;
                                                     ~~~~~~~~~~~

node_modules/@types/mongoose/index.d.ts:332:27 - error TS2694: Namespace '"/home/t00rk/mif/entities/node_modules/mongodb/mongodb.ts34"' has no exported member 'SessionOptions'.

332         options?: mongodb.SessionOptions,
                              ~~~~~~~~~~~~~~

node_modules/@types/mongoose/index.d.ts:390:31 - error TS2694: Namespace '"/home/t00rk/mif/entities/node_modules/mongodb/mongodb.ts34"' has no exported member 'CollectionCreateOptions'.

390             options?: mongodb.CollectionCreateOptions,
                                  ~~~~~~~~~~~~~~~~~~~~~~~

node_modules/@types/mongoose/index.d.ts:398:30 - error TS2694: Namespace '"/home/t00rk/mif/entities/node_modules/mongodb/mongodb.ts34"' has no exported member 'CollectionCreateOptions'.

398             options: mongodb.CollectionCreateOptions,
                                 ~~~~~~~~~~~~~~~~~~~~~~~

node_modules/@types/mongoose/index.d.ts:612:31 - error TS2694: Namespace '"/home/t00rk/mif/entities/node_modules/mongodb/mongodb.ts34"' has no exported member 'SessionOptions'.

612             options?: mongodb.SessionOptions,
                                  ~~~~~~~~~~~~~~

node_modules/@types/mongoose/index.d.ts:1943:51 - error TS2551: Property 'ObjectID' does not exist on type 'typeof import("/home/t00rk/mif/entities/node_modules/mongodb/mongodb.ts34")'. Did you mean 'ObjectId'?

1943         type ObjectIdConstructor = typeof mongodb.ObjectID & {
                                                       ~~~~~~~~

node_modules/@types/mongoose/index.d.ts:1944:44 - error TS2694: Namespace '"/home/t00rk/mif/entities/node_modules/mongodb/mongodb.ts34"' has no exported member 'ObjectID'.

1944             (s?: string | number): mongodb.ObjectID;
                                                ~~~~~~~~

node_modules/@types/mongoose/index.d.ts:1949:44 - error TS2694: Namespace '"/home/t00rk/mif/entities/node_modules/mongodb/mongodb.ts34"' has no exported member 'ObjectID'.

1949         interface ObjectId extends mongodb.ObjectID {}
                                                ~~~~~~~~

node_modules/@types/mongoose/index.d.ts:2245:50 - error TS2694: Namespace '"/home/t00rk/mif/entities/node_modules/mongodb/mongodb.ts34"' has no exported member 'FindAndModifyWriteOpResultObject'.

2245             callback?: (error: any, doc: mongodb.FindAndModifyWriteOpResultObject<DocType | null>, result: any) => void,
                                                      ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

node_modules/@types/mongoose/index.d.ts:2246:26 - error TS2694: Namespace '"/home/t00rk/mif/entities/node_modules/mongodb/mongodb.ts34"' has no exported member 'FindAndModifyWriteOpResultObject'.

2246         ): Query<mongodb.FindAndModifyWriteOpResultObject<DocType | null>> & QueryHelpers;
                              ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

node_modules/@types/mongoose/index.d.ts:2277:48 - error TS2694: Namespace '"/home/t00rk/mif/entities/node_modules/mongodb/mongodb.ts34"' has no exported member 'FindAndModifyWriteOpResultObject'.

2277             callback?: (err: any, doc: mongodb.FindAndModifyWriteOpResultObject<DocType>, res: any) => void,
                                                    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

node_modules/@types/mongoose/index.d.ts:2278:26 - error TS2694: Namespace '"/home/t00rk/mif/entities/node_modules/mongodb/mongodb.ts34"' has no exported member 'FindAndModifyWriteOpResultObject'.

2278         ): Query<mongodb.FindAndModifyWriteOpResultObject<DocType>> & QueryHelpers;
                              ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

node_modules/@types/mongoose/index.d.ts:2289:48 - error TS2694: Namespace '"/home/t00rk/mif/entities/node_modules/mongodb/mongodb.ts34"' has no exported member 'FindAndModifyWriteOpResultObject'.

2289             callback?: (err: any, doc: mongodb.FindAndModifyWriteOpResultObject<DocType | null>, res: any) => void,
                                                    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

node_modules/@types/mongoose/index.d.ts:2290:26 - error TS2694: Namespace '"/home/t00rk/mif/entities/node_modules/mongodb/mongodb.ts34"' has no exported member 'FindAndModifyWriteOpResultObject'.

2290         ): Query<mongodb.FindAndModifyWriteOpResultObject<DocType | null>> & QueryHelpers;
                              ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

node_modules/@types/mongoose/index.d.ts:2531:62 - error TS2694: Namespace '"/home/t00rk/mif/entities/node_modules/mongodb/mongodb.ts34"' has no exported member 'WriteOpResult'.

2531         remove(callback?: (err: any) => void): Query<mongodb.WriteOpResult["result"]> & QueryHelpers;
                                                                  ~~~~~~~~~~~~~

node_modules/@types/mongoose/index.d.ts:2535:26 - error TS2694: Namespace '"/home/t00rk/mif/entities/node_modules/mongodb/mongodb.ts34"' has no exported member 'WriteOpResult'.

2535         ): Query<mongodb.WriteOpResult["result"]> & QueryHelpers;
                              ~~~~~~~~~~~~~

node_modules/@types/mongoose/index.d.ts:3391:42 - error TS2694: Namespace '"/home/t00rk/mif/entities/node_modules/mongodb/mongodb.ts34"' has no exported member 'BulkWriteOpResultObject'.

3391             cb?: (err: any, res: mongodb.BulkWriteOpResultObject) => void,
                                              ~~~~~~~~~~~~~~~~~~~~~~~

node_modules/@types/mongoose/index.d.ts:3392:28 - error TS2694: Namespace '"/home/t00rk/mif/entities/node_modules/mongodb/mongodb.ts34"' has no exported member 'BulkWriteOpResultObject'.

3392         ): Promise<mongodb.BulkWriteOpResultObject>;
                                ~~~~~~~~~~~~~~~~~~~~~~~

node_modules/@types/mongoose/index.d.ts:3395:31 - error TS2694: Namespace '"/home/t00rk/mif/entities/node_modules/mongodb/mongodb.ts34"' has no exported member 'CollectionBulkWriteOptions'.

3395             options?: mongodb.CollectionBulkWriteOptions,
                                   ~~~~~~~~~~~~~~~~~~~~~~~~~~

node_modules/@types/mongoose/index.d.ts:3396:28 - error TS2694: Namespace '"/home/t00rk/mif/entities/node_modules/mongodb/mongodb.ts34"' has no exported member 'BulkWriteOpResultObject'.

3396         ): Promise<mongodb.BulkWriteOpResultObject>;
                                ~~~~~~~~~~~~~~~~~~~~~~~

node_modules/@types/mongoose/index.d.ts:3399:30 - error TS2694: Namespace '"/home/t00rk/mif/entities/node_modules/mongodb/mongodb.ts34"' has no exported member 'CollectionBulkWriteOptions'.

3399             options: mongodb.CollectionBulkWriteOptions,
                                  ~~~~~~~~~~~~~~~~~~~~~~~~~~

node_modules/@types/mongoose/index.d.ts:3400:41 - error TS2694: Namespace '"/home/t00rk/mif/entities/node_modules/mongodb/mongodb.ts34"' has no exported member 'BulkWriteOpResultObject'.

3400             cb: (err: any, res: mongodb.BulkWriteOpResultObject) => void,
                                             ~~~~~~~~~~~~~~~~~~~~~~~

node_modules/@types/mongoose/index.d.ts:3475:44 - error TS2694: Namespace '"/home/t00rk/mif/entities/node_modules/mongodb/mongodb.ts34"' has no exported member 'CollectionCreateOptions'.

3475         createCollection(options?: mongodb.CollectionCreateOptions, cb?: (err: any) => void): Promise<void>;
                                                ~~~~~~~~~~~~~~~~~~~~~~~

node_modules/@types/mongoose/index.d.ts:3620:48 - error TS2694: Namespace '"/home/t00rk/mif/entities/node_modules/mongodb/mongodb.ts34"' has no exported member 'FindAndModifyWriteOpResultObject'.

3620             callback?: (err: any, res: mongodb.FindAndModifyWriteOpResultObject<T | null>) => void,
                                                    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

node_modules/@types/mongoose/index.d.ts:3621:26 - error TS2694: Namespace '"/home/t00rk/mif/entities/node_modules/mongodb/mongodb.ts34"' has no exported member 'FindAndModifyWriteOpResultObject'.

3621         ): Query<mongodb.FindAndModifyWriteOpResultObject<T | null>> & QueryHelpers;
                              ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

node_modules/@types/mongoose/index.d.ts:3646:48 - error TS2694: Namespace '"/home/t00rk/mif/entities/node_modules/mongodb/mongodb.ts34"' has no exported member 'FindAndModifyWriteOpResultObject'.

3646             callback?: (err: any, res: mongodb.FindAndModifyWriteOpResultObject<T | null>) => void,
                                                    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

node_modules/@types/mongoose/index.d.ts:3647:26 - error TS2694: Namespace '"/home/t00rk/mif/entities/node_modules/mongodb/mongodb.ts34"' has no exported member 'FindAndModifyWriteOpResultObject'.

3647         ): Query<mongodb.FindAndModifyWriteOpResultObject<T | null>> & QueryHelpers;
                              ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

node_modules/@types/mongoose/index.d.ts:3679:48 - error TS2694: Namespace '"/home/t00rk/mif/entities/node_modules/mongodb/mongodb.ts34"' has no exported member 'FindAndModifyWriteOpResultObject'.

3679             callback?: (err: any, res: mongodb.FindAndModifyWriteOpResultObject<T>) => void,
                                                    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

node_modules/@types/mongoose/index.d.ts:3680:26 - error TS2694: Namespace '"/home/t00rk/mif/entities/node_modules/mongodb/mongodb.ts34"' has no exported member 'FindAndModifyWriteOpResultObject'.

3680         ): Query<mongodb.FindAndModifyWriteOpResultObject<T>> & QueryHelpers;
                              ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

node_modules/@types/mongoose/index.d.ts:3685:48 - error TS2694: Namespace '"/home/t00rk/mif/entities/node_modules/mongodb/mongodb.ts34"' has no exported member 'FindAndModifyWriteOpResultObject'.

3685             callback?: (err: any, res: mongodb.FindAndModifyWriteOpResultObject<T | null>) => void,
                                                    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

node_modules/@types/mongoose/index.d.ts:3686:26 - error TS2694: Namespace '"/home/t00rk/mif/entities/node_modules/mongodb/mongodb.ts34"' has no exported member 'FindAndModifyWriteOpResultObject'.

3686         ): Query<mongodb.FindAndModifyWriteOpResultObject<T | null>> & QueryHelpers;
                              ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

node_modules/@types/mongoose/index.d.ts:3746:48 - error TS2694: Namespace '"/home/t00rk/mif/entities/node_modules/mongodb/mongodb.ts34"' has no exported member 'FindAndModifyWriteOpResultObject'.

3746             callback?: (err: any, doc: mongodb.FindAndModifyWriteOpResultObject<T | null>, res: any) => void,
                                                    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

node_modules/@types/mongoose/index.d.ts:3747:26 - error TS2694: Namespace '"/home/t00rk/mif/entities/node_modules/mongodb/mongodb.ts34"' has no exported member 'FindAndModifyWriteOpResultObject'.

3747         ): Query<mongodb.FindAndModifyWriteOpResultObject<T | null>> & QueryHelpers;
                              ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

node_modules/@types/mongoose/index.d.ts:3770:48 - error TS2694: Namespace '"/home/t00rk/mif/entities/node_modules/mongodb/mongodb.ts34"' has no exported member 'FindAndModifyWriteOpResultObject'.

3770             callback?: (err: any, doc: mongodb.FindAndModifyWriteOpResultObject<T | null>, res: any) => void,
                                                    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

node_modules/@types/mongoose/index.d.ts:3771:26 - error TS2694: Namespace '"/home/t00rk/mif/entities/node_modules/mongodb/mongodb.ts34"' has no exported member 'FindAndModifyWriteOpResultObject'.

3771         ): Query<mongodb.FindAndModifyWriteOpResultObject<T | null>> & QueryHelpers;
                              ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

node_modules/@types/mongoose/index.d.ts:3797:48 - error TS2694: Namespace '"/home/t00rk/mif/entities/node_modules/mongodb/mongodb.ts34"' has no exported member 'FindAndModifyWriteOpResultObject'.

3797             callback?: (err: any, doc: mongodb.FindAndModifyWriteOpResultObject<T>, res: any) => void,
                                                    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

node_modules/@types/mongoose/index.d.ts:3798:26 - error TS2694: Namespace '"/home/t00rk/mif/entities/node_modules/mongodb/mongodb.ts34"' has no exported member 'FindAndModifyWriteOpResultObject'.

3798         ): Query<mongodb.FindAndModifyWriteOpResultObject<T>> & QueryHelpers;
                              ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

node_modules/@types/mongoose/index.d.ts:3809:48 - error TS2694: Namespace '"/home/t00rk/mif/entities/node_modules/mongodb/mongodb.ts34"' has no exported member 'FindAndModifyWriteOpResultObject'.

3809             callback?: (err: any, doc: mongodb.FindAndModifyWriteOpResultObject<T | null>, res: any) => void,
                                                    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

node_modules/@types/mongoose/index.d.ts:3810:26 - error TS2694: Namespace '"/home/t00rk/mif/entities/node_modules/mongodb/mongodb.ts34"' has no exported member 'FindAndModifyWriteOpResultObject'.

3810         ): Query<mongodb.FindAndModifyWriteOpResultObject<T | null>> & QueryHelpers;
                              ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

node_modules/@types/mongoose/index.d.ts:3920:26 - error TS2694: Namespace '"/home/t00rk/mif/entities/node_modules/mongodb/mongodb.ts34"' has no exported member 'DeleteWriteOpResultObject'.

3920         ): Query<mongodb.DeleteWriteOpResultObject["result"] & { deletedCount?: number }> & QueryHelpers;
                              ~~~~~~~~~~~~~~~~~~~~~~~~~

node_modules/@types/mongoose/index.d.ts:3924:26 - error TS2694: Namespace '"/home/t00rk/mif/entities/node_modules/mongodb/mongodb.ts34"' has no exported member 'DeleteWriteOpResultObject'.

3924         ): Query<mongodb.DeleteWriteOpResultObject["result"] & { deletedCount?: number }> & QueryHelpers;
                              ~~~~~~~~~~~~~~~~~~~~~~~~~

node_modules/@types/mongoose/index.d.ts:3929:26 - error TS2694: Namespace '"/home/t00rk/mif/entities/node_modules/mongodb/mongodb.ts34"' has no exported member 'DeleteWriteOpResultObject'.

3929         ): Query<mongodb.DeleteWriteOpResultObject["result"] & { deletedCount?: number }> & QueryHelpers;
                              ~~~~~~~~~~~~~~~~~~~~~~~~~

node_modules/@types/mongoose/index.d.ts:3933:26 - error TS2694: Namespace '"/home/t00rk/mif/entities/node_modules/mongodb/mongodb.ts34"' has no exported member 'DeleteWriteOpResultObject'.

3933         ): Query<mongodb.DeleteWriteOpResultObject["result"] & { deletedCount?: number }> & QueryHelpers;
                              ~~~~~~~~~~~~~~~~~~~~~~~~~

node_modules/@types/mongoose/index.d.ts:3938:26 - error TS2694: Namespace '"/home/t00rk/mif/entities/node_modules/mongodb/mongodb.ts34"' has no exported member 'DeleteWriteOpResultObject'.

3938         ): Query<mongodb.DeleteWriteOpResultObject["result"] & { deletedCount?: number }> & QueryHelpers;
                              ~~~~~~~~~~~~~~~~~~~~~~~~~

node_modules/mongodb/mongodb.ts34.d.ts:4532:89 - error TS2344: Type 'Flatten<TSchema[key]>' does not satisfy the constraint '{ _id?: any; }'.
  Type 'unknown' is not assignable to type '{ _id?: any; }'.
    Type 'Flatten<TSchema[KeysOfAType<TSchema, readonly any[] | undefined>]>' is not assignable to type '{ _id?: any; }'.
      Type 'unknown' is not assignable to type '{ _id?: any; }'.
        Type 'Flatten<TSchema[NonNullable<TSchema[keyof TSchema]> extends readonly any[] | undefined ? keyof TSchema : never]>' is not assignable to type '{ _id?: any; }'.
          Type 'unknown' is not assignable to type '{ _id?: any; }'.
            Type 'Flatten<TSchema[keyof TSchema]>' is not assignable to type '{ _id?: any; }'.
              Type 'unknown' is not assignable to type '{ _id?: any; }'.
                Type 'Flatten<TSchema[string | number | symbol]>' is not assignable to type '{ _id?: any; }'.
                  Type 'unknown' is not assignable to type '{ _id?: any; }'.
                    Type 'Flatten<TSchema[string]> | Flatten<TSchema[number]> | Flatten<TSchema[symbol]>' is not assignable to type '{ _id?: any; }'.
                      Type 'Flatten<TSchema[string]>' is not assignable to type '{ _id?: any; }'.
                        Type 'unknown' is not assignable to type '{ _id?: any; }'.

4532     readonly [key in KeysOfAType<TSchema, ReadonlyArray<any> | undefined>]?: OptionalId<Flatten<TSchema[key]>> | AddToSetOperators<Array<OptionalId<Flatten<TSchema[key]>>>>;
                                                                                             ~~~~~~~~~~~~~~~~~~~~~

node_modules/mongodb/mongodb.ts34.d.ts:4532:149 - error TS2344: Type 'Flatten<TSchema[key]>' does not satisfy the constraint '{ _id?: any; }'.

4532     readonly [key in KeysOfAType<TSchema, ReadonlyArray<any> | undefined>]?: OptionalId<Flatten<TSchema[key]>> | AddToSetOperators<Array<OptionalId<Flatten<TSchema[key]>>>>;

```

## Code Example

I have not yet coded anything related to typegoose.

It happens when I use `npm run clean && tsc` with an empty `index.ts` file.

## Do you know *why* it happens?

*no*

---

**Comment by hasezoey** [comment]
could you provide your package.json, because this seems like wrong versions or an package not being installed?

**Comment by T00rk** [comment]
@hasezoey 

**package.json**
```json
{
  "name": "@mif/entities",
  "version": "0.3.0",
  "description": "",
  "main": "./build/index.js",
  "types": "./build/index.d.ts",
  "files": [
    "build/**/*"
  ],
  "scripts": {
    "clean": "del ./build/*",
    "build": "npm run clean && tsc",
    "pub": "npm version patch && npm run build && npm publish"
  },
  "keywords": [],
  "author": "T00rk",
  "license": "ISC",
  "devDependencies": {
    "@types/mongoose": "^5.10.4",
    "del-cli": "^3.0.0",
    "typescript": "^3.9.9"
  },
  "dependencies": {
    "@typegoose/typegoose": "^7.4.1",
    "@types/node": "^16.4.3",
    "moment": "^2.29.1",
    "mongoose": "^5.10.18",
    "mongoose-update-if-current": "^1.4.0",
    "ts-node-dev": "^1.0.0-pre.44"
  }
}
```

**tsconfig.json**
```json
{
  "compilerOptions": {
    "target": "ES2018",
    "module": "commonjs", 
    "declaration": true,
    "outDir": "./build",
    "strict": true,
    "esModuleInterop": true,
    "experimentalDecorators": true, 
    "emitDecoratorMetadata": true, 
    "forceConsistentCasingInFileNames": true
  }
}

**Comment by hasezoey** [comment]
from what i can tell, it look like you basically have the same problem as in #581, only difference being that your versions are too new (in the mentioned issue the versions were too old)

(if you look in the lockfile you would see that your versions are in an unsupported range for `@types/mongoose` and `mongoose`)

TL;DR:
change `@types/mongoose`'s version to `~5.10.x`
change `mongoose`'s version to `5.10.18` (hard, no `^` or `~`)
update typegoose to latest version, currently being `7.6.3` (not just the lockfile)

[Look here for why](https://typegoose.github.io/typegoose/docs/guides/faq#why-is-74x-constrained-to-mongoose-51018)

**Comment by T00rk** [comment]
@hasezoey I just tried. It doesn't work

```json
...
"devDependencies": {
    "@types/mongoose": "~5.10.x",
    "del-cli": "^3.0.0",
    "typescript": "^3.9.9"
  },
  "dependencies": {
    "@typegoose/typegoose": "^7.6.3",
    "@types/node": "^16.4.3",
    "moment": "^2.29.1",
    "mongoose": "5.10.18",
    "mongoose-update-if-current": "^1.4.0",
    "ts-node-dev": "^1.0.0-pre.44"
  }
```

**Comment by hasezoey** [comment]
what is your package manager and did it update the lockfile to the correct versions?
(and do you have any thing else installed that would require higher versions, like plugins?)

**Comment by T00rk** [comment]
I'm using npm. I deleted the lockfile and `npm_modules` directory before changing my `package.json`.
I thought `mongoose-update-if-current` could influence mongoose version so I deleted it. 
Result was the same.

**Comment by hasezoey** [comment]
now that i look at the errors again, it looks like mongodb's types are somehow wrong, do you by chance (even implicitly) have mongodb `4.x` installed, when yes, try to force it down to an `3.x` version (inside the lockfile if implicit)

Reasoning: this seems like errors from mongodb 4.x, because since then it got rewritten in typescript and provides its own types (so the types and names are different)

**Comment by hasezoey** [comment]
@T00rk does this issue still exist in typegoose 8.0.0 with mongoose 5.13+?

**Comment by T00rk** [comment]
@hasezoey it seems to work nice. I've just tried to compile a model and it worked.
Thank you
